dot/core/service.go

```diff
@@ -89,7 +90,14 @@ func NewService(cfg *Config) (*Service, error) {
 		return nil, ErrNilStorageState
 	}
 
-	if cfg.Network == nil {
+	var isNilNetwork bool
+	switch cfg.Network.(type) {
+	case *network.Service:
+		isNilNetwork = cfg.Network == (*network.Service)(nil)
+	default:
```

Comment:
maybe put a comment why this is needed

Reply:
What are the other type implementations for `cfg.Network`? :thinking:

Reply:
there aren't any, so maybe we don't need the default case, hmm

Reply:
Yeah but we will we actually ever have a mock network typed nil ? 🤔